src/Nethermind/Nethermind.Serialization.Rlp/TxDecoder.cs

```diff
@@ -103,7 +112,23 @@ protected TxDecoder(bool lazyHash = true)
                 rlpStream.Check(lastCheck);
             }
 
-            if (transactionSequence.Length <= TxDecoder.MaxDelayedHashTxnSize && _lazyHash)
+            if ((rlpBehaviors & RlpBehaviors.InNetworkForm) == RlpBehaviors.InNetworkForm && transaction.HasNetworkForm)
+            {
+                switch (transaction.Type)
+                {
+                    case TxType.Blob:
+                        DecodeShardBlobNetworkPayload(transaction, rlpStream, rlpBehaviors);
+                        break;
+                }
+
+                if ((rlpBehaviors & RlpBehaviors.AllowExtraBytes) != RlpBehaviors.AllowExtraBytes)
+                {
+                    rlpStream.Check(networkWrapperCheck);
+                }
+
+                transaction.Hash = CalculateHashForNetworkPayloadForm(transaction.Type, transactionSequence);
+            }
+            else if (transactionSequence.Length <= TxDecoder.MaxDelayedHashTxnSize && _lazyHash)
```

Comment:
@benaadams will this constant work well for blobs? TxDecoder.MaxDelayedHashTxnSize? @flcl42 ?

Reply:
Yeah, will mean it will likely calculate the Keccak upfront when the txn hits the pool before deciding if it can discard it for some reason (so higher cpu)

If it becomes issue can pull forward some of the early txn rejections from the pool to decoding as they can be detected in early part of header

Reply:
Shard Blob Tx part used for hashing is not much bigger than for others, adds ~40..140 bytes.